plugins/functions/math.c

```diff
@@ -0,0 +1,127 @@
+/* -*- c-basic-offset: 2 -*- */
+/*
+  Copyright(C) 2017 Brazil
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License version 2.1 as published by the Free Software Foundation.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifdef GRN_EMBEDDED
+#  define GRN_PLUGIN_FUNCTION_TAG functions_math
+#endif
+
+#include <groonga/plugin.h>
+
+#include <stdlib.h>
+
+static grn_obj *
+func_math_abs(grn_ctx *ctx, int n_args, grn_obj **args,
+              grn_user_data *user_data)
+{
+  grn_obj *number;
+  grn_obj *grn_abs_number;
+
+  if (n_args != 1) {
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "math_abs(): wrong number of arguments (%d for 1)",
+                     n_args);
+    return NULL;
+  }
+
+  number = args[0];
+  if (!(number->header.type == GRN_BULK &&
+        grn_type_id_is_number_family(ctx, number->header.domain) &&
+        number->header.domain != GRN_DB_FLOAT)) {
```

Comment:
Why is `GRN_DB_FLOAT` out of support? If we don't support `GRN_DB_FLOAT`, we can't use `math_abs()` for `_score`.